src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnection.cs

```diff
@@ -1448,12 +1448,10 @@ private Task WriteAsciiStringAsync(string s, bool async)
             int offset = _writeOffset;
             if (s.Length <= _writeBuffer.Length - offset)
             {
-                byte[] writeBuffer = _writeBuffer;
-                foreach (char c in s)
-                {
-                    writeBuffer[offset++] = (byte)c;
-                }
-                _writeOffset = offset;
+                OperationStatus operationStatus = Ascii.FromUtf16(s, _writeBuffer.AsSpan(offset), out _, out int bytesWritten);
```

Comment:
The loop above in WriteStringAsync:
https://github.com/dotnet/runtime/pull/75012/files#diff-595f700a4c83b735d856fe14b5ebe8c7f7c18ae6e586adc8b398a997a0678274L1391-L1398
can also be replaced.